soci/soci_index.go

```diff
@@ -542,6 +561,13 @@ func (b *IndexBuilder) build(ctx context.Context, img images.Image, buildCfg bui
 					// index layers must be in some deterministic order
 					// actual layer order used for historic consistency
 					sociLayersDesc[i] = desc
+					if toc != nil && enablePrefetch {
+						builtZtocs = append(builtZtocs, &ztocWithLayer{
```

Comment:
nit: can this directly insert into the correct index (like we do for `sociLayerDesc`)? I think we could get rid of index from `ztocWithLayer` and avoid sorting later.